### DISCUSSION TITLE
Connecting requires multiple attempts ...

### DISCUSSION THREAD

**Discussion by JonFP** [codefori/vscode-ibmi#786] [General]
I have three systems that I connect to.  One of them (and only one so far) often requires multiple connection attempts before it "takes". One student in a class on Tuesday had to press connect 5 times before it worked. I've had it take 4 goes and this morning it took 3.

Is there a log somewhere that might show what is causing this ?

Anyone else seeing this behaviour?  I'm seeing it from Macs and windows machines and there is nothing special about the system's configuration.

**Comment by worksofliam** [comment]
@JonFP What message are you getting when it fails to connect? I've seen this before and I've heard it's not specific to Code for IBM i.

**Reply by JonFP** [reply]
<img width="562" alt="image" src="https://user-images.githubusercontent.com/13474625/181601558-a654e9a8-5bbe-4e9f-95d1-d9ab80f81c27.png">

It comes back almost immediately.   To get that image I disconnected and then reconnected. Took four times to connect but other times it hits first time.

**Reply by worksofliam** [reply]
What happens if you try and connect via SSH from a shell? A similar message?

**Reply by JonFP** [reply]
I have never had any issues from Mac terminal if that's what you mean.

**Reply by worksofliam** [reply]
Yes, that's what I mean. What a very off situation. I am asking around for you!

**Comment by JonFP** [comment]
Thanks.

Now that I think back it is possible that it has happened once or twice with a regular ssh connection - but I simply put it down to me screwing up and tried again.  I'll keep an eye out for it.


Jon Paris

***@***.***

**Comment by chrjorgensen** [comment]
@JonFP As a guy using SSH everywhere, your problem to me sounds like a server or network problem - "no response from server". Also since it is only one server out of three, it points against that specific server.

I would recommend the following troubleshooting procedure the next time you encounter this issue:
1. Immediately after having this issue the first time in Code for IBM i, open a terminal on the same workstation and do a regular SSH against the same server. If this also fails, you can eliminate VS Code. You could having the terminal ready before starting a connection to the server in Code for IBM i.
2. Also try SSH from a terminal on a different workstation - to eliminate the workstation with Code for IBM i.
3. Ping the server to see if any packet drops occur. It will by default run forever on a Mac (and on Linux).
4. Talk to the server administrator to see if any performance issue or configuration on the server could be the issue. E.g. the SSH could be unable to start a new job for your connection, if the subsystem has reached some limit.

Not knowing the system landscape and network where the server is located, it is difficult to pinpoint the cause. But as I said in the beginning, it points against the server.

SSH is a brilliant and secure tool for connection between two computers, and even Windows Server now has a OpenSSH server (and client) available. I even use it between IBM i partitions to run commands on server two from server one. I have never experience any issue like yours in Code for IBM i - so I don't think the problem lies there...

Hope this helps you in finding the cause of your problems.

Best regards,
Christian
